modules/openthread/CMakeLists.txt

```diff
@@ -163,6 +163,11 @@ endforeach()
 string(REPLACE " " ";" OT_PARAM_LIST " ${CONFIG_OPENTHREAD_CUSTOM_PARAMETERS}")
 target_compile_definitions(ot-config INTERFACE ${OT_PARAM_LIST})
 
+# Since Mbed TLS 3.1.0 MBEDTLS_SSL_EXPORT_KEYS was removed as build symbol and
+# it's always assumed to be enabled. Corresponding kconfig was removed from
+# Zephyr as well, but OpenThread code still uses it, so we add it here.
+target_compile_definitions(ot-config INTERFACE -DMBEDTLS_SSL_EXPORT_KEYS)
```

Comment:
Hmm... Wouldn't the proper fix be to make the OpenThread code that relies on this macro to not anymore? If you don't want to do this yourself pinging relevant people could be a good idea.

Reply:
That was an option, but on one hand this requires opening a separate PR on the OpenThread repo and on the other one I don't know what's the preferred version of Mbed TLS to be used with OpenThread. `MBEDTLS_SSL_EXPORT_KEYS` was only removed after Mbed TLS 3.1.0, so if you build OpenThread with previous versions that build symbol is still available.
This fix looks easier and safer IMO. Wdyt?

Reply:
Ah so the code that makes use of this macro in OpenThread doesn't live in the main Zephyr repo? In any case this is fine for me. Might just be worth pinging relevant people to make them aware of this.